Add Close Tab Button

TimChen44 opened the pull request:
Provide a tab closure event. After defining the event, the tab displays a closure button, and the control itself does not perform any logic. The closure function is implemented by the user themselves.
![tab](https://github.com/radzenhq/radzen-blazor/assets/7581981/28eb9561-d23e-41f9-bc53-2247292d8219)

akorchev commented on Radzen.Blazor/RadzenTabs.razor.cs:
```diff
@@ -71,6 +71,13 @@ public partial class RadzenTabs : RadzenComponent
+        /// </summary>
+        /// <value>The close button callback.</value>
+        [Parameter]
+        public EventCallback<RadzenTabsItem> TabCloseClick { get; set; }
```
Please rename it to just `TabClose`

akorchev commented on Radzen.Blazor/RadzenTabsItem.razor.cs:
```diff
@@ -138,6 +138,14 @@ async Task OnClick()
 
+        async Task OnTabCloseClick()
+        {
+            if (Tabs.TabCloseClick.HasDelegate)
```
There is no need for the if check here.

akorchev commented on Radzen.Blazor/RadzenTabsItem.razor:
```diff
@@ -16,6 +16,14 @@
+                {
+                    <span class="rz-ml-05">
+                        <RadzenButton Click="@OnTabCloseClick" class="rz-border-radius" Icon="close" Variant="Variant.Text"
+                                      ButtonStyle="ButtonStyle.Dark" Shade="Shade.Darker" Size="ButtonSize.ExtraSmall"
```
Having both ButtonStyle.Dark and Shade.Darker may look odd in a dark theme. What if those are not set at all?

TimChen44 commented:
Fixed the above issue and added the close button display mode.
The FREE style has been modified. I don't know how to modify the PRO style, so I didn't make any changes.

![GIF 2023-9-13 22-43-26](https://github.com/radzenhq/radzen-blazor/assets/7581981/2ea3a72c-c6e0-47cd-ae37-0ef449fc62c3)

akorchev commented:
We discussed this internally and decided we should implement something else - a Template property of RadzenTabsItem which would allow adding arbitrary content - buttons, badge etc. The template would display inside the `<a>` element. Would you like to try implementing that?

TimChen44 commented:
The common scenarios here are the close button and badge, but there may be other buttons in some complex scenarios. Supporting templates is a good idea, and I can try to improve it.

akorchev commented:
It turns out there is a Template property already 🤣 Perhaps we can just make a demo which implements the closing with buttons that are defined in the template. Also the common UX pattern is to display the close button on hover only.

TimChen44 commented:
One problem with existing templates is that users need to create everything themselves, which increases the complexity of user usage for commonly used scenarios.

TimChen44 commented:
Added BadgeText attribute, I feel the effect is quite good.

akorchev commented:
Hi @TimChen44,

Thank you for your efforts however we decided to not continue with this PR. We prefer to use the Template for such purposes.
